test/XRP/xrp-client-integration.test.ts

```diff
@@ -229,4 +229,71 @@ describe('XrpClient Integration Tests', function (): void {
       AccountRootFlags.checkFlag(AccountRootFlags.LSF_DEPOSIT_AUTH, flags!),
     )
   })
+
+  it('Authorize Sending Account - failure on authorizing self', async function (): Promise<
```

Comment:
there are definitely ways to test this stuff by querying the ledger, but we don't have gRPC endpoints for the objects and i need and i didn't want to block on figuring this out. 

that work is tracked in https://app.asana.com/0/1190398011050043/1196153592884122/f, and will clean up these tests